### PR TITLE
kubevirt: package upgrade to 1.2.0

### PR DESCRIPTION
**Pull request by elainezhao96**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

kubevirt  package upgrade to 1.2.0
updated patch and spec file based on https://code.opensuse.org/package/kubevirt/blob/master/f/kubevirt.spec#_287

###### Change Log  

- upgrade kubevirt to 1.2.0

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

triggered build  https://dev.azure.com/mariner-org/mariner/_build/results?buildId=546557&view=results

### CODE REVIEW

**Comment by elainezhao96** [comment]
the kubevirt packages installed fine and libvirt related pkgs failed to install https://dev.azure.com/mariner-org/mariner/_build/results?buildId=526707&view=logs&j=8fc237db-d9de-5497-d5bc-7b55eebfb9b4&t=95de2199-5f24-5bb6-ce75-dd26f99e8922 I believe it is at least validated for this kubevirt upgrade. Please let me know if further action is required

File: SPECS/kubevirt/Allocate-2-cpu-for-the-emulator-thread.patch [hunk 1]
```diff
@@ -15,20 +15,26 @@ diff --git a/pkg/virt-controller/services/renderresources.go b/pkg/virt-controll
...
  		if cpu.IsolateEmulatorThread {
--			emulatorThreadCPU := resource.NewQuantity(1, resource.BinarySI)
-+			emulatorThreadCPU := resource.NewQuantity(2, resource.BinarySI)
- 			limits := renderer.calculatedLimits[k8sv1.ResourceCPU]
- 			limits.Add(*emulatorThreadCPU)
- 			renderer.vmLimits[k8sv1.ResourceCPU] = limits
+-			emulatorThreadCPUs := resource.NewQuantity(1, resource.BinarySI)
++			emulatorThreadCPUs := resource.NewQuantity(2, resource.BinarySI)
+
+			limits := renderer.calculatedLimits[k8sv1.ResourceCPU]
```
Comment (romoh) [inline SPECS/kubevirt/Allocate-2-cpu-for-the-emulator-thread.patch:30]: Not sure about this, we only need to make the quality(2)

Comment (romoh) [reply SPECS/kubevirt/Allocate-2-cpu-for-the-emulator-thread.patch:30]: Looks like we shouldn't need the patch at all given this check `emulatorThreadCompleteToEvenParityAnnotationExists` , the right fix would be to add new annotations

Comment (elainezhao96) [reply SPECS/kubevirt/Allocate-2-cpu-for-the-emulator-thread.patch:30]: to confirm, I dont need to do any other fix other than removing this patch?  FYI @kanikanema

Comment (romoh) [reply SPECS/kubevirt/Allocate-2-cpu-for-the-emulator-thread.patch]: correct. We should note that though for CAPKV migration to the new version.

**Review by romoh — APPROVED** [review]

File: SPECS/kubevirt/kubevirt.spec [hunk 2]
```diff
@@ -18,8 +18,8 @@
 
 Summary:        Container native virtualization
 Name:           kubevirt
-Version:        0.59.0
-Release:        14%{?dist}
+Version:        1.2.0
```
Comment (bfjelds) [inline SPECS/kubevirt/kubevirt.spec:21]: just double checking, there are no changes made to the spec other than version??  that seems rare.

File: SPECS/kubevirt/kubevirt.spec [hunk 3]
```diff
@@ -30,8 +30,7 @@ Source1:        disks-images-provider.yaml
 # Nexus team needs these to-be-upstreamed patches for the operator Edge to work
 # correctly.
 Patch0:         Cleanup-housekeeping-cgroup-on-vm-del.patch
-Patch1:         Allocate-2-cpu-for-the-emulator-thread.patch
```
Comment (bfjelds) [inline SPECS/kubevirt/kubevirt.spec:33]: was the allocate patch upstreamed?  or just not needed?

Comment (bfjelds) [reply SPECS/kubevirt/kubevirt.spec:33]: sounds like kubevirt handled this through annotations, patch no longer needed.

File: SPECS/kubevirt/kubevirt.spec [hunk 4]
```diff
@@ -18,28 +18,29 @@
...
 %global debug_package %{nil}
-BuildRequires:  glibc-devel
-BuildRequires:  glibc-static >= 2.38-3%{?dist}
-BuildRequires:  golang
+BuildRequires:  glibc-devel-static
 BuildRequires:  golang-packaging
 BuildRequires:  pkgconfig
 BuildRequires:  rsync
 BuildRequires:  sed
+BuildRequires:  golang(API) >= 1.21
```
Comment (vinceaperri) [inline SPECS/kubevirt/kubevirt.spec:43]: Should this just be `BuildRequires:  golang >= 1.21`?

**Review by vinceaperri — CHANGES_REQUESTED** [review]

File: SPECS/kubevirt/kubevirt.signatures.json [hunk 5]
```diff
@@ -1,6 +1,9 @@
 {
   "Signatures": {
-    "disks-images-provider.yaml": "6da2f333ad97e2c6e16272bec70a1747d11505377e5aa27ccae550f00b368c43",
-    "kubevirt-0.59.0.tar.gz": "4f9557e9102163e2d9c3ca237cb40cc2e88f05d03b4e0d97745b2a09e5ab3a78"
+    "disks-images-provider.yaml": "c0b58839b33c93e3d111274edf4fe1b6736c2ebe3e8a4b581b9ae298fe29ea6d",
+    "kubevirt-1.2.0.tar.gz": "da63ee052939a68e54dca37084234ebd7a86233f2087bc13cb7d63fc24d356e3",
+    "kubevirt-rpmlintrc": "16311e9a9550813799516f1871376d95151c8fd4abf4e5a5a568f91fa7c9e10a",
+    "kubevirt_containers_meta": "a7899c8992be6387148a3f3f5d0bc0464d3259a9272602e5efbc42c09a604295",
+    "kubevirt_containers_meta.service": "a38783b77ebea6546a0c94e7ca5deb5f1a4d890633d754f11f12a85aad27de4e"
```
Comment (vinceaperri) [inline SPECS/kubevirt/kubevirt.signatures.json:7]: These files do not exist so shouldn't be in here. Or maybe they are missing from the PR?

```
kubevirt-rpmlintrc
kubevirt_containers_meta
kubevirt_containers_meta.service
```

File: SPECS/kubevirt/Hotplug_detach_grace_period.patch [hunk 6]
```diff
@@ -2,7 +2,7 @@ diff --git a/pkg/virt-launcher/virtwrap/manager.go b/pkg/virt-launcher/virtwrap/
 index 63d8d43d2..0e608c4bb 100644
 --- a/pkg/virt-launcher/virtwrap/manager.go
 +++ b/pkg/virt-launcher/virtwrap/manager.go
-@@ -97,6 +97,11 @@ const (
```
Comment (vinceaperri) [inline SPECS/kubevirt/Hotplug_detach_grace_period.patch:5]: Where is this patch from?

File: SPECS/kubevirt/kubevirt.spec [hunk 7]
```diff
@@ -18,28 +18,29 @@
...
-Release:        14%{?dist}
+Version:        1.2.0
+Release:        1%{?dist}
 License:        ASL 2.0
 Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
 Group:          System/Management
 URL:            https://github.com/kubevirt/kubevirt
 Source0:        https://github.com/kubevirt/kubevirt/archive/refs/tags/v%{version}.tar.gz#/%{name}-%{version}.tar.gz
-Source1:        disks-images-provider.yaml
```
Comment (vinceaperri) [inline SPECS/kubevirt/kubevirt.spec:29]: Where are these sources being used?

**Review by vinceaperri — CHANGES_REQUESTED** [review]

File: SPECS/kubevirt/kubevirt.spec [hunk 8]
```diff
@@ -18,28 +18,29 @@
...
 # correctly.
 Patch0:         Cleanup-housekeeping-cgroup-on-vm-del.patch
-Patch1:         Allocate-2-cpu-for-the-emulator-thread.patch
-Patch2:         Hotplug_detach_grace_period.patch
+Patch1:         Hotplug_detach_grace_period.patch
 %global debug_package %{nil}
-BuildRequires:  glibc-devel
-BuildRequires:  glibc-static >= 2.38-3%{?dist}
-BuildRequires:  golang
+BuildRequires:  glibc-devel-static
```
[the remark this answers is not in the document]
Comment (vinceaperri) [reply SPECS/kubevirt/kubevirt.spec:38]: I would both packages to have the same version constraint as well.

File: SPECS/kubevirt/kubevirt.spec [hunk 9]
```diff
@@ -204,13 +353,38 @@ install -p -m 0644 cmd/virt-handler/nsswitch.conf %{buildroot}%{_datadir}/kube-v
...
+%files -n obs-service-kubevirt_containers_meta
+%license LICENSE
+%doc README.md
+%dir %{_prefix}/lib/obs
+%{_prefix}/lib/obs/service
 
 %changelog
+* Wed Mar 13 2024 Elaine Zhao <elainezhao@microsoft.com> - 1.2.0-1
+- Bump package version to 1.2.0 from 0.59.0
+
```
Comment (sharath-srikanth-chellappa) [inline SPECS/kubevirt/kubevirt.spec:387]: Can you add the link/URL where you found the requirement to add the `pr-helper-conf`, `manifests` and other new packages along with the host of changes required to the SPEC file.

File: SPECS/libvirt/libvirt.spec [hunk 10]
```diff
@@ -688,7 +688,7 @@ Requires: gzip
 Requires: bzip2
 Requires: lzop
 Requires: xz
-Requires: swtpm-tools
+#Requires: swtpm-tools
```
Comment (sharath-srikanth-chellappa) [inline SPECS/libvirt/libvirt.spec:691]: Could you also provide documentation as to why swtpm-tools is not required anymore and why this change is being made.

Please add it to the changelog for this current spec file since you are making a change here

File: SPECS/libvirt/libvirt.spec
Comment (sharath-srikanth-chellappa) [inline SPECS/libvirt/libvirt.spec:1]: You may want to increment the `Release:` in this file since you are making changes to the packages here

**Comment by sharath-srikanth-chellappa** [comment]
There are 2 CVE patches that have been merged on `2.0` last month. @elainezhao96 can you check if these 2 CVE Patches need to be cherry-picked onto this branch which is going to be merged to `3.0-dev`

File: SPECS/libvirt/libvirt.spec (hunk 10 above)
Comment (elainezhao96) [reply SPECS/libvirt/libvirt.spec:691]: this was temporarily commented out for me the trigger libvirt build, will revert back.

File: SPECS/kubevirt/kubevirt.signatures.json (hunk 5 above)
Comment (elainezhao96) [reply SPECS/kubevirt/kubevirt.signatures.json:7]: they are missing from pr. Will add them

**Comment by elainezhao96** [comment]
> There are 2 CVE patches that have been merged on `2.0` last month. @elainezhao96 can you check if these 2 CVE Patches need to be cherry-picked onto this branch which is going to be merged to `3.0-dev`

these no longer affect 1.2.0

File: SPECS/kubevirt/kubevirt.spec (hunk 7 above)
Comment (elainezhao96) [reply SPECS/kubevirt/kubevirt.spec:29]: those are used in the %install section

Comment (elainezhao96) [reply SPECS/kubevirt/kubevirt.spec:29]: actually, seems kubevirt_containers_meta is in older versions of kubevirt.spec but it was not included previously. so I guess it will be safe to remove them

File: SPECS/kubevirt/kubevirt.spec [hunk 11]
```diff
@@ -18,24 +18,26 @@
...
 License:        ASL 2.0
 Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
 Group:          System/Management
 URL:            https://github.com/kubevirt/kubevirt
 Source0:        https://github.com/kubevirt/kubevirt/archive/refs/tags/v%{version}.tar.gz#/%{name}-%{version}.tar.gz
-Source1:        disks-images-provider.yaml
+Source1:        kubevirt_containers_meta
+Source2:        kubevirt_containers_meta.service
+Source3:        disks-images-provider.yaml
```
Comment (kanikanema) [inline SPECS/kubevirt/kubevirt.spec:31]: We can drop this.

File: SPECS/kubevirt/kubevirt.spec [hunk 12]
```diff
@@ -95,20 +116,111 @@ Group:          System/Packages
...
 Group:          System/Packages
 
 %description    tests
 The package provides Kubevirt end-to-end tests.
 
+%package -n     obs-service-kubevirt_containers_meta
+Summary:        Kubevirt containers meta information (build service)
+Group:          System/Packages
+
+%description -n obs-service-kubevirt_containers_meta
```
Comment (kanikanema) [inline SPECS/kubevirt/kubevirt.spec:146]: We don't need the containers_meta packages (and associated files) for mariner. Suggest removing it and add a comment here instead so future upgrades can refer it.

**Review by kanikanema — COMMENTED** [review]
LGTM, some minor comments.

**Comment by elainezhao96** [comment]
@kanikanema Thanks for the comments! But I'm hitting error as I added some packages in the .spec:
Missing build-id in /usr/src/azl/BUILDROOT/kubevirt-1.2.0-1.azl3.x86_64/usr/bin/virtctl" 
Missing build-id in /usr/src/azl/BUILDROOT/kubevirt-1.2.0-1.azl3.x86_64/usr/bin/virt-api"
Missing build-id in /usr/src/azl/BUILDROOT/kubevirt-1.2.0-1.azl3.x86_64/usr/bin/virt-controller"
Missing build-id in /usr/src/azl/BUILDROOT/kubevirt-1.2.0-1.azl3.x86_64/usr/bin/virt-exportproxy"

do you recall how virt-launcher build id was specified?

**Comment by kanikanema** [comment]
> @kanikanema Thanks for the comments! One more question: I'm hitting error in https://dev.azure.com/mariner-org/mariner/_build/results?buildId=530815&view=results as I added some packages in the .spec: Missing build-id in /usr/src/azl/BUILDROOT/kubevirt-1.2.0-1.azl3.x86_64/usr/bin/virtctl" Missing build-id in /usr/src/azl/BUILDROOT/kubevirt-1.2.0-1.azl3.x86_64/usr/bin/virt-api" Missing build-id in /usr/src/azl/BUILDROOT/kubevirt-1.2.0-1.azl3.x86_64/usr/bin/virt-controller" Missing build-id in /usr/src/azl/BUILDROOT/kubevirt-1.2.0-1.azl3.x86_64/usr/bin/virt-exportproxy"
> 
> do you recall how virt-launcher build id was specified?

Hi Elaine, Can you confirm which version of the patchset did you run this build with?
AFAICS, the build-id missing are RPM warnings, the main errors seem to be this:

Directory not found: /usr/src/azl/BUILDROOT/kubevirt-1.2.0-1.azl3.x86_64/usr/share/kube-virt/manifests"
File not found: /usr/src/azl/BUILDROOT/kubevirt-1.2.0-1.azl3.x86_64/usr/share/kube-virt/manifests/release"

I don't recall doing anything special to include build-ids for the virt-* binaries. I can't spot anything the spec diff either that would be relevant to the warning.

File: SPECS/kubevirt/kubevirt.spec [hunk 13]
```diff
@@ -15,27 +15,27 @@
 # Please submit bugfixes or comments via https://bugs.opensuse.org/
 #
 
+%define _missing_build_ids_terminate_build 0
```
Comment (kanikanema) [inline SPECS/kubevirt/kubevirt.spec:18]: Where is this variable being used?

**Review by kanikanema — COMMENTED** [review]
LGTM.

**Review by romoh — APPROVED** [review]

File: SPECS/kubevirt/kubevirt.spec (hunk 13 above)
Comment (bfjelds) [reply SPECS/kubevirt/kubevirt.spec:18]: i think related to build errors like this:

`Missing build-id in /usr/src/azl/BUILDROOT/kubevirt-1.2.0-1.azl3.x86_64/usr/bin/virtctl`

File: SPECS/libvirt/libvirt.spec
Comment (sharath-srikanth-chellappa) [reply SPECS/libvirt/libvirt.spec:1]: I still see changes to the SPEC file here. You should be incrementing `Release:` if you are making changes

File: SPECS/kubevirt/kubevirt.spec [hunk 14]
```diff
@@ -204,13 +256,24 @@ install -p -m 0644 cmd/virt-handler/nsswitch.conf %{buildroot}%{_datadir}/kube-v
...
 %license LICENSE
 %doc README.md
 %dir %{_datadir}/kube-virt
 %{_bindir}/virt-tests
 
 %changelog
+* Wed Mar 13 2024 Elaine Zhao <elainezhao@microsoft.com> - 1.2.0-1
+- Bump package version to 1.2.0 from 0.59.0
+- Referenced kubevirt.spec from https://code.opensuse.org/package/kubevirt/blob/master/f/kubevirt.spec
+
```
Comment (bfjelds) [inline SPECS/kubevirt/kubevirt.spec:276]: maybe add note about build-id issue here?

File: SPECS/kubevirt/kubevirt.signatures.json (hunk 5 above)
Comment (sharath-srikanth-chellappa) [reply SPECS/kubevirt/kubevirt.signatures.json]: These files are still not added to the PR

**Review by sharath-srikanth-chellappa — CHANGES_REQUESTED** [review]

**Comment by bfjelds** [comment]
the PR comment has notes about buddy builds that required commenting out swtpm, which shouldn't be needed anymore.  have buddy builds been run with the current commit?

**Review by bfjelds — APPROVED** [review]

**Review by sharath-srikanth-chellappa — APPROVED** [review]
LGTM

**Review by vinceaperri — APPROVED** [review]
